_rules/printable-characters-shortcut-ffbc54.md

```diff
@@ -0,0 +1,803 @@
+---
+id: ffbc54
+name: No keyboard shortcuts use only printable characters
+rule_type: composite
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, then there is a mechanism to disable the shortcut, or to remap the shortcut to use one or more non-printable characters keys, or the shortcut for a user interface component is only available when that component has focus.
+accessibility_requirements:
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_rules:
+  - 670a30
+  - aa8b52
+  - 1e9941
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] with at least one [keyboard shortcut][] that requires pressing only [printable character][] keys to trigger an action on a [user interface component][].
+
+## Expectation
+
+For the test target, the outcome of at least one of the following rules is passed:
+
+- [Printable keys only shortcut can be disabled](https://act-rules.github.io/rules/670a30)
+- [Printable keys only shortcut can be remapped](https://act-rules.github.io/rules/aa8b52) 
+- [Printable key shortcut is only active when component has focus](https://act-rules.github.io/rules/1e9941)
```

Comment:
I'm now having questions about the core of the composite rule 😱 

Consider something like that:
```
onload="
  registerShortcut( {focusOnly: true, shortcutKey: "a"});
  registerShortcut( {shortcutKey: "b", id: "b"});
  registerShortcut( {shortcutKey: "c", id: "c"});"
…
<input type="checkbox" onclick="toggleDisabled('b', !this.checked)">
<input type="checkbox" onclick="toggleModifier('c', !this.checked)">
```
This has a single UI component with 3 single key shortcuts. This does not pass any to the atomic rules:
1. The "disable" rule fails because shortcuts "a" and "c" cannot be disabled.
2. The "remap" rule fails because shortcuts "a" and "b" cannot be remapped.
3. The "focus" rule fails because shortcuts "b" and "c" are active even without focus.

Because all atomic rules fail, the composite rule also fails but this does not fail WCAG because each individual shortcut is good.

---
Because of the structure of the composite rules that we have (can only look at the result of the atomic rules, not at details), this will  stay as long as the applicability is on the full page…
The atomic rules are actually checking whether **ALL** shortcuts on the page are implementing a **single** method of making them good. Thus the composite rule can only get as input the answers to questions like "do **all** shortcuts on the page implement method A?" It is not possible for the composite rule to pass on a page where shortcuts implements different methods because the composite rule never sees the result of individual shortcuts.

---
This kinda shows that the rules must target individual shortcuts one way or another… Even targeting individual UI components is not enough given the example above.